internal/staging/stdscript/scriptv0.go

```diff
@@ -714,3 +714,122 @@ func MultiSigScriptV0(threshold int, pubKeys ...[]byte) ([]byte, error) {
 
 	return builder.Script()
 }
+
+// AtomicSwapDataPushesV0 houses the data pushes found in hash-based atomic swap
+// contracts using version 0 scripts.
+type AtomicSwapDataPushesV0 struct {
+	RecipientHash160 [20]byte
+	RefundHash160    [20]byte
+	SecretHash       [32]byte
+	SecretSize       int64
+	LockTime         int64
+}
+
+// ExtractAtomicSwapDataPushesV0 returns the data pushes from an atomic swap
+// contract using version 0 scripts if it is one.  It will return nil otherwise.
+//
+// NOTE: Atomic swaps are not considered standard script types by the dcrd
+// mempool policy and should be used with P2SH.  The atomic swap format is also
+// expected to change to use a more secure hash function in the future.
+func ExtractAtomicSwapDataPushesV0(redeemScript []byte) *AtomicSwapDataPushesV0 {
+	// Local constants for convenience.
+	const (
+		maxMathOpCodeLen = txscript.MathOpCodeMaxScriptNumLen
+		maxCltvLen       = txscript.CltvMaxScriptNumLen
+	)
+
+	// An atomic swap is of the form:
+	//  IF
+	//   SIZE <secret size> EQUALVERIFY SHA256 <32-byte secret> EQUALVERIFY DUP
```

Comment:
I believe that `<32-byte secret>` should say  `<32-byte secret hash>` as it is the SHA256 hash of the data.

Reply:
Agreed.  Added.